### DISCUSSION TITLE
SSR ReferenceError w/ react-sigma

### DISCUSSION THREAD

**Discussion by cchin25** [vercel/next.js#73861] [Help] [answered]
### Summary

I'm using `react-sigma` in next.js 15 w/ the app directory. I noticed during dev that the following error is thrown.

```
 ⨯ ReferenceError: WebGL2RenderingContext is not defined
    at [project]/app/my-graph.tsx [app-ssr] (ecmascript) (app/my-graph.tsx:2:0)
    at [project]/app/page.tsx [app-ssr] (ecmascript) (app/page.tsx:2:0)
  1 | "use client";
> 2 | import { useLoadGraph } from "@react-sigma/core";
  3 | import { MultiDirectedGraph } from "graphology";
  4 | import { useEffect } from "react";
  5 | {
```

I have "use client" specified in both my page.tsx and the my-graph.tsx file. Runtime everything appears to work correctly. Is it safe to ignore the error? It builds and there's no runtime client side errors.

I've also attempted using dynamic imports but it doesn't seem to make a difference because it seems the issue is while it parses the `useLoadGraph` hook during the import?  I'm stumped if anyone has any thoughts.

**Comment by icyJoseph** [comment]
> I have "use client" specified in both my page.tsx and the my-graph.tsx file. Runtime everything appears to work correctly. Is it safe to ignore the error? It builds and there's no runtime client side errors.

Yeah, `use client` doesn't mean browser only. Client components are SSR'd, https://github.com/reactwg/server-components/discussions/4

What could happen here is that, they are doing some top level checks on objects that are only available in a browser.

Did you pass `{ ssr: false }` to the `next/dynamic` import?

**Reply by icyJoseph** [reply, marked as answer]
Yeah, `ssr: false` should do the trick, as per this issue from their repo: https://github.com/sim51/react-sigma/issues/66#issuecomment-2114182704

**Reply by icyJoseph** [reply]
There's also this react-sigma core documentation bit, https://sim51.github.io/react-sigma/docs/faq/#is-it-possible-to-use-this-project-with-nextjs-, which I don't like that much, but hey, it uses `ssr: false`

**Reply by cchin25** [reply]
I did see the docs on the FAQ which is what initially led me to try it. I didn't see the issue reported but will compare against what I had. Also thanks for the gaeron link, haven't seen that post before, appreciate it.

**Reply by cchin25** [reply]
Okay, so updating similar to how it was mentioned in the issue thread also worked for me and I didn't need to dynamically import the `SigmaContainer` which I was initially trying to do.
